pandas/tests/window/test_timeseries_window.py

```diff
@@ -690,3 +706,34 @@ def test_rolling_cov_offset(self):
 
         expected2 = ss.rolling(3, min_periods=1).cov()
         tm.assert_series_equal(result, expected2)
+
+    def test_rolling_on_decreasing_index(self):
+        # GH-19248
+        index = [
+            Timestamp("20190101 09:00:00"),
+            Timestamp("20190101 09:00:02"),
+            Timestamp("20190101 09:00:03"),
+            Timestamp("20190101 09:00:05"),
+            Timestamp("20190101 09:00:06"),
+        ]
+
+        df = DataFrame({"column": [3, 4, 4, 2, 1]}, index=reversed(index))
+        result = df.rolling("2s").min()
+        expected = DataFrame(
+            {"column": [3.0, 3.0, 3.0, 2.0, 1.0]}, index=reversed(index)
```

Comment:
It seems to me that the 2 seconds time window is not respected, but the minimum is taken over the whole dataframe. To my understanding, the excpected results would be [3, 4, 2, 1, 1] (closed interval limits) or [3, 4, 2, 2, 1] (open interval end). Even if you consider the next 2 seconds in time order, i.e. the other way round, you would have [3, 3, 4, 2, 1].

Reply:
Same issue found here. By using df.rolling().apply(func), we can print the window inside func and it is the while dataframe till current location.

Reply:
You're right, the expected result is wrong. I debugged it a bit and found out that `VariableWindowIndexer` indeed works only if the index is monotonic and increasing. I will create another PR to finally fix this.